.github/workflows/main.yml

```diff
@@ -0,0 +1,48 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  pull_request:
+
+jobs:
+  tox:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.5, 3.6, 3.7, pypy3, 3.8]
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set Up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Get pip cache dir
+        id: pip-cache
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+
+      - name: pip cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('pyproject.toml', 'setup.py', 'setup.cfg') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Install
+        run: |
+          pip install tox
+
+      - name: tox
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: tox -e py-test-twlatest,py-test-twtrunk,py-linters,publishcov,release
```

Comment:
think that py-linters can be executed as part of a separate workflow and it should be enough to run it only on Linux.
Same for the release environment.

My problem with tox in general is that it is very verbose / noisy.

For example, if a tests fails, with the current setup I will have to scroll 2500 lines and somewhere in the middle I will see the error details.

What I do for my projects, is send all important output to a separate file and then print that file as a separate step.
That separate file will contain only the important test failure details.

For tox, as a workaround I am running tox --notest first with full verbosity to create the environmet and have a trace of the exact versions used and then I run tox -qqq to just run the tests and see the errors in a dedicated step.